src/paging.rs

```diff
@@ -617,6 +624,31 @@ impl<T: Translation> PageTableWithLevel<T> {
         Ok(())
     }
 
+    /// Walks a range of page table entries and passes each one to a caller provided function
+    /// If the range is not aligned to block boundaries, it will be expanded.
+    fn walk_range<F>(
+        &self,
+        translation: &T,
+        range: &MemoryRegion,
+        f: &mut F,
+    ) -> Result<(), MapError>
+    where
+        F: FnMut(&MemoryRegion, &Descriptor, usize),
+    {
+        let level = self.level;
+        for chunk in range.split(level) {
+            // VA range passed to the updater is aligned to block boundaries, as that region will
+            // be affected by changes to the entry.
```

Comment:
What changes?

Reply:
Hmm this is still not quite right - let me resolve that properly